### PR TITLE
Refactor native build to a shell script

### LINKED ISSUE #2598: [FEA] Make the native build process readable for devs without Maven experience
**Is your feature request related to a problem? Please describe.**
Whenever there is an issue with the native build, there is an additional hurdle of understanding this repo's build flow to the rapidsai/cudf developers.
It is complex enough and on top of that it is obscured in multiple Maven antrun plugin executions that are hard to read especially for users unfamiliar with Maven.

**Describe the solution you'd like**
Either consolidate all the cmake invocations in a single shell script and the script logs all its parameters so it is easy to run without Maven in the build container
or have some dry-run option that generates the build script.

**Describe alternatives you've considered**
We can document in more detail how to extract the steps from Maven debug output and reproduce each step in the build container without Maven. But I expect it to be harder than generating a bash script to run by a rapidsai engineer

**Additional context**
Breaking changes in libcudf 

cc @vyasr

### PR DESCRIPTION
**Pull request by jlowe**
Fixes #2598.  Refactors the native build into a shell script capable of being run standalone.  The shell script is configured to log its commands to make it easier to see in a buildlog how cmake is being invoked across the phases of the native build.

### CODE REVIEW

**Comment by jlowe** [comment]
build

**Review by jihoonson — DISMISSED** [review]

**Comment by gerashegalov** [comment]
This LGTM mostly. 

Ideally I would like to be able to tell the external engineer a single command to rerun in the Docker container.

I put forward my suggestions as a PR to your PR branch for consideration 
https://github.com/jlowe/spark-rapids-jni/pull/1/files?diff=unified&w=1

The idea is that we reproduce the native build with a single command after dropping into the build Docker container
```
$ ./build/run-in-docker 
```
Run
```
$ REUSE_ENV=true ./build/buildcpp.sh 
```

This instruction appears in the log, we can also document it in the dev documentation.

File: build/buildcpp.sh [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+#
+
+#
+# Script to build native code in cudf and spark-rapids-jni
+#
+
+set -ex
+
+# Environment variables to control the build
+PROJECT_BASE_DIR=${PROJECT_BASE_DIR:-$(realpath $(dirname $0)/..)}
```
[the remark this answers is not in the document]
Comment (jlowe) [reply build/buildcpp.sh:42]: BUILD_FAULTINJ is already there?

**Comment by jlowe** [comment]
build

File: build/buildcpp.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (gerashegalov) [reply build/buildcpp.sh:42]: correct, however, I also thought we should move the caveat logic for ARM builds in here. Thanks for accommodating.

File: pom.xml [hunk 2]
```diff
@@ -451,12 +448,15 @@
                   <env key="CUDF_PATH" value="${cudf.path}"/>
                   <env key="CUDF_PIN_PATH" value="${cudf.pin.path}"/>
                   <env key="CUDF_USE_PER_THREAD_DEFAULT_STREAM" value="${CUDF_USE_PER_THREAD_DEFAULT_STREAM}"/>
+                  <env key="FROM_MAVEN" value="true"/>
```
Comment (gerashegalov) [inline pom.xml:451]: good idea

**Review by gerashegalov — APPROVED** [review]
LGTM

**Pull request merged**